spec/features/pages/event_component_spec.rb

```diff
@@ -0,0 +1,66 @@
+    create(:page_component, page: @page, component: @e3, position: 4)
+
+    @h2 = create(:page_header2_component,subtopic_title: "Visible Past Events")
+    create(:page_component, page: @page, component: @h2, position: 5)
+    @e4 = create(:page_event_component, title: "past event1", start_date: today - 2.days, end_date: today - 1.days, hide_once_passed: false)
+    create(:page_component, page: @page, component: @e4, position: 6)
+    @e5 = create(:page_event_component, title: "past event2", start_date: today - 2.days, end_date: today - 1.days, hide_once_passed: false)
+    create(:page_component, page: @page, component: @e5, position: 7)
+    @e6 = create(:page_event_component, title: "past event3", start_date: today - 2.days, end_date: today - 1.days, hide_once_passed: false)
+    create(:page_component, page: @page, component: @e6, position: 8)
+
+    @h3 = create(:page_header2_component,subtopic_title: "Hidden Past Events")
+    create(:page_component, page: @page, component: @h3, position: 9)
+    @e7 = create(:page_event_component, title: "past event4", start_date: today - 2.days, end_date: today - 1.days, hide_once_passed: true)
+    create(:page_component, page: @page, component: @e7, position: 10)
+    @e8 = create(:page_event_component, title: "past event5", start_date: today - 2.days, end_date: today - 1.days, hide_once_passed: true)
+    create(:page_component, page: @page, component: @e8, position: 11)
+    @e9 = create(:page_event_component, title: "past event6", start_date: today - 2.days, end_date: today - 1.days, hide_once_passed: true)
+    create(:page_component, page: @page, component: @e9, position: 12)
+
+    user = create(:user)
+    login_as(user, scope: :user, run_callbacks: false)
+  end
+
+  it 'displays events correctly with pagination functionality' do
+    visit 'programming/ruby-rocks'
+
+    expect(page).to have_content(@h1.subtopic_title)
+    expect(page).to have_content(@e1.title)
+    expect(page).to have_content(@e2.title)
+    expect(page).not_to have_content(@e3.title)
+    page.has_link?('Load more', href: /events-0=2/)
+    find('a', text: 'Load more', match: :first).click
+    expect(page).to have_content(@e3.title)
+
+    expect(page).to have_content(@h2.subtopic_title)
+    expect(page).to have_content(@e4.title)
+    expect(page).to have_content(@e5.title)
+    expect(page).not_to have_content(@e6.title)
+    page.has_link?('Load more', href: /events-1=2/)
+    find('a', text: 'Load more', match: :first).click
+    expect(page).to have_content(@e6.title)
+
+    expect(page).to have_content(@h3.subtopic_title)
+    expect(page).to have_content("Please check back for more events soon!")
+    expect(page).not_to have_content(@e7.title)
+    expect(page).not_to have_content(@e8.title)
+    expect(page).not_to have_content(@e9.title)
+    expect(page).not_to have_css('.dm-load-more-events-2-btn-container')
+  end
+end
+
```

Comment:
This feature test is very verbose with the fixture creation, but the nuance with adding page_components with correct positions in a way that will mimic practical use was a bit difficult to do in a DRY way so I opted for straight-forward.